tests/sites_tests/tests.py

```diff
@@ -86,6 +86,43 @@ def test_get_current_site_no_site_id(self):
         site = get_current_site(request)
         self.assertEqual(site.name, "example.com")
 
+    @override_settings(SITE_ID='', ALLOWED_HOSTS=['example.com', 'example.net'])
+    def test_get_current_site_no_site_id_and_handle_port_fallback(self):
+        request = HttpRequest()
+
+        s1 = self.site
+        s2 = Site.objects.create(domain='example.com:80', name='example.com:80')
+
+        # Test for host header without port:
+        request.META = {'HTTP_HOST': 'example.com'}
+        site = get_current_site(request)
+        self.assertIsInstance(site, Site)
+        self.assertEqual(site, s1)
+
+        # Test for host header with port - match, no fallback without port:
+        request.META = {'HTTP_HOST': 'example.com:80'}
+        site = get_current_site(request)
+        self.assertIsInstance(site, Site)
+        self.assertEqual(site, s2)
+
+        # Test for host header with port - no match, fallback without port:
+        request.META = {'HTTP_HOST': 'example.com:81'}
+        site = get_current_site(request)
+        self.assertIsInstance(site, Site)
+        self.assertEqual(site, s1)
+
+        # Test for host header with non-matching domain:
+        request.META = {'HTTP_HOST': 'example.net'}
+        self.assertRaises(ObjectDoesNotExist, get_current_site, request)
+
+        # Ensure domain for RequestSite always matches host header:
+        with self.modify_settings(INSTALLED_APPS={'remove': 'django.contrib.sites'}):
+            for host in 'example.com', 'example.com:80':
```

Comment:
Can you please unfold this loop. It's hard to check what actually failed if one item in the list fails.